var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -175,7 +175,7 @@ class Sundials(CMakePackage, CudaPackage, ROCmPackage):
     depends_on('lapack',              when='+lapack')
     depends_on('suite-sparse',        when='+klu')
     depends_on('petsc+mpi',           when='+petsc')
-    depends_on('hypre+mpi',           when='+hypre')
+    depends_on('hypre@:2.22.0+mpi',   when='@:5.7.0 +hypre')
```

Comment:
We want to still depend on `hypre+mpi` when > 5.7.0 (i.e. @develop and future releases), so I think we want this instead:

```suggesting
depends_on('hypre+mpi',                  when='+hypre')
depends_on('hypre@:2.22.0+mpi',   when='@:5.7.0 +hypre')
```

Reply:
Sorry @balos1 but when I do that I get "An unsatisfiable version constraint has been detected for spec".
I would need, as you are asking for develop and future, something like
```
    depends_on('hypre+mpi',           when='@5.7.1: +hypre')
    depends_on('hypre@:2.22.0+mpi',   when='@:5.7.0 +hypre')
```